### PR TITLE
Adds boilerplate project setup with log function

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This starts with a log function that looks like this in WebAssembly text format:

```webassembly
  (import "http" "log"
    (func $http.log (param $ptr i32) (param $size i32)))
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package http_wasm
```
Comment (mathetake) [inline main.go]: let's name it as a single word following the convention
```suggestion
package httpwasm
```

Comment (codefromthecrypt) [reply main.go]: sure I'll change that all around

File: imports.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+//go:build tinygo.wasm
+
+package httpwasm
+
+//go:wasm-module httpwasm
```
Comment (codefromthecrypt) [inline imports.go]: ps this has the WebAssembly import module as "httpwasm", but maybe it is nicer as "http_wasm"? wdyt? In any case glad it isn't "env"!

File: .github/workflows/commit.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: "Set up TinyGo"
+        run: |  # Installing via curl so commands are similar on OS/x
+          tinygo_version=${{ matrix.tinygo-version }}
+          curl -sSL https://github.com/tinygo-org/tinygo/releases/download/v${tinygo_version}/tinygo${tinygo_version}.linux-amd64.tar.gz | sudo tar -C /usr/local -xzf -
+          echo "TINYGOROOT=/usr/local/tinygo" >> $GITHUB_ENV
+          echo "/usr/local/tinygo/bin" >> $GITHUB_PATH
+
+      - name: "Set up wasmtime"
+        run: |  # TinyGo currently requires wasmtime to execute `tingo test -target=wasi ...`
```
Comment (anuraaga) [inline .github/workflows/commit.yaml]:
```suggestion
        run: |  # TinyGo currently requires wasmtime to execute `tinygo test -target=wasi ...`
```

File: imports.go (hunk 2 above)
Comment (anuraaga) [inline imports.go]: Since it's already wasm, maybe it should be just `http`?

Comment (codefromthecrypt) [reply imports.go]: very good idea

Comment (codefromthecrypt) [reply imports.go]: http is far better choice thanks @anuraaga!

**Review by anuraaga — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
revised description with the toehold import description cc @Taction 

next, I'll do the fasthttp part

Thanks @anuraaga and @mathetake for the naming polish!

**Pull request merged**